### ISSUE TITLE
Dokku on Ubuntu 18.04 Multipass VM won't deploy any app

### ISSUE THREAD

**Issue by Ci7rix** [dokku/dokku#3951]
## Description of problem

I can't get Dokku to work on a Ubuntu 18.04 VM created with Multipass on macOS. Everytime I deploy an app, I get : `Failed to download a Ruby executable for bootstrapping!`

### How reproducible

Everytime

### Steps to Reproduce

1. Create a VM with Multipass : `multipass launch --name "dokku" --mem 2048M --disk 10G bionic`

2. Login in the VM : `multipass shell dokku`

3. Install Dokku following the Quick-start Instructions (curl|bash)

4. Deploying Ruby Getting Started following the official documentation

#### Actual Results

App is not deployed, ruby is not downloaded

```
ruby-getting-started - [master] » git push dokku master
Énumération des objets: 406, fait.
Décompte des objets: 100% (406/406), fait.
Compression par delta en utilisant jusqu'à 4 fils d'exécution
Compression des objets: 100% (226/226), fait.
Écriture des objets: 100% (406/406), 85.13 Kio | 85.13 Mio/s, fait.
Total 406 (delta 154), réutilisés 406 (delta 154), réutilisés du pack 0
remote: Resolving deltas: 100% (154/154), done.
-----> Cleaning up...
-----> Building ruby-getting-started from herokuish...
-----> Setting config vars
       CURL_CONNECT_TIMEOUT:  90
-----> Setting config vars
       CURL_TIMEOUT:  600
-----> Adding BUILD_ENV to build environment...
-----> Ruby app detected
       Failed to download a Ruby executable for bootstrapping!
       
       This is most likely a temporary internal error. If the problem
       persists, make sure that you are not running a custom or forked
       version of the Heroku Ruby buildpack which may need updating.
remote: /tmp/buildpacks/01_buildpack-ruby/bin/compile: line 11: /tmp/tmp.vqg0JGmzYg/bin/ruby: No such file or directory
To dokku.test:ruby-getting-started
 ! [remote rejected] master -> master (pre-receive hook declined)
error: impossible de pousser des références vers 'dokku@dokku.test:ruby-getting-started'
```

Gist with with trace turned on [here](https://gist.github.com/Ci7rix/776bcd0a653f87d8eb48ea317c1351b9)

#### Expected Results

App getting deployed

## Environment Information

```
-----> uname: Linux dokku 4.15.0-96-generic #97-Ubuntu SMP Wed Apr 1 03:25:46 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
                     total        used        free      shared  buff/cache   available
       Mem:           1993         192         222          11        1578        1637
       Swap:             0           0           0
-----> docker version:
       Client: Docker Engine - Community
        Version:           19.03.8
        API version:       1.40
        Go version:        go1.12.17
        Git commit:        afacb8b7f0
        Built:             Wed Mar 11 01:25:46 2020
        OS/Arch:           linux/amd64
        Experimental:      false

       Server: Docker Engine - Community
        Engine:
         Version:          19.03.8
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.12.17
         Git commit:       afacb8b7f0
         Built:            Wed Mar 11 01:24:19 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.2.13
         GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
        runc:
         Version:          1.0.0-rc10
         GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info:
WARNING: No swap limit support
       Client:
        Debug Mode: true

       Server:
        Containers: 5
         Running: 1
         Paused: 0
         Stopped: 4
        Images: 9
        Server Version: 19.03.8
        Storage Driver: overlay2
         Backing Filesystem: <unknown>
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
        runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 4.15.0-96-generic
        Operating System: Ubuntu 18.04.4 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 1
        Total Memory: 1.947GiB
        Name: dokku
        ID: DTZX:ZGYM:YC5O:J24J:VTAO:6RKS:QV22:LJ6T:DMFI:K2AU:YC2Q:GV2Z
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false

-----> sigil version: 0.5.0
-----> herokuish version:
       herokuish: 0.5.11
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v214
         heroku-buildpack-nodejs    v170
         heroku-buildpack-clojure   v84
         heroku-buildpack-python    v167
         heroku-buildpack-java      v66
         heroku-buildpack-gradle    v31
         heroku-buildpack-scala     v87
         heroku-buildpack-play      v26
         heroku-buildpack-php       v173
         heroku-buildpack-go        v139
         buildpack-nginx            v12
-----> dokku version: dokku version 0.20.4
-----> dokku plugins:
       plugn: 0.3.2
         00_dokku-standard    0.20.4 enabled    dokku core standard plugin
         20_events            0.20.4 enabled    dokku core events logging plugin
         app-json             0.20.4 enabled    dokku core app-json plugin
         apps                 0.20.4 enabled    dokku core apps plugin
         builder-dockerfile   0.20.4 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.20.4 enabled    dokku core builder-herokuish plugin
         buildpacks           0.20.4 enabled    dokku core buildpacks plugin
         certs                0.20.4 enabled    dokku core certificate management plugin
         checks               0.20.4 enabled    dokku core checks plugin
         common               0.20.4 enabled    dokku core common plugin
         config               0.20.4 enabled    dokku core config plugin
         docker-options       0.20.4 enabled    dokku core docker-options plugin
         domains              0.20.4 enabled    dokku core domains plugin
         enter                0.20.4 enabled    dokku core enter plugin
         git                  0.20.4 enabled    dokku core git plugin
         logs                 0.20.4 enabled    dokku core logs plugin
         network              0.20.4 enabled    dokku core network plugin
         nginx-vhosts         0.20.4 enabled    dokku core nginx-vhosts plugin
         plugin               0.20.4 enabled    dokku core plugin plugin
         postgres             1.11.2 enabled    dokku postgres service plugin
         proxy                0.20.4 enabled    dokku core proxy plugin
         ps                   0.20.4 enabled    dokku core ps plugin
         repo                 0.20.4 enabled    dokku core repo plugin
         resource             0.20.4 enabled    dokku core resource plugin
         scheduler-docker-local 0.20.4 enabled    dokku core scheduler-docker-local plugin
         shell                0.20.4 enabled    dokku core shell plugin
         ssh-keys             0.20.4 enabled    dokku core ssh-keys plugin
         storage              0.20.4 enabled    dokku core storage plugin
         tags                 0.20.4 enabled    dokku core tags plugin
         tar                  0.20.4 enabled    dokku core tar plugin
         trace                0.20.4 enabled    dokku core trace plugin
```

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?:

Installed with the quickstart curl|bash on a Ubuntu 18.04 VM on Multipass on macOS.

**Comment by josegonzalez** [comment]
Funny that in trying to install within multipass, I get network errors pulling the herokuish image. I feel as though I had crappy initial networking the last time I tried to use multipass to test the ubuntu microk8s distro...

I've recreated your install via multipass on my mac, and am not able to replicate the issue when deploying the [ruby-getting-started](https://github.com/heroku/ruby-getting-started) app. Any other things you think I can try?

**Comment by Ci7rix** [comment]
Hello thank you for your answer!

I did some more tests.

 I started by trying on an image in 16.04 always on the same host, no problem, everything was working correctly. On the same machine in version 20.04, impossible to make it work (apt installation method) as for version 18.04.

Then I tried the installation on a freshly installed mac with the 18.04 version of the image again no problem.

I finally found that dnsmasq linked to Valet Laravel was running in the background, after having disabled it, everything was working correctly.

What I find hard to understand is how is it possible that with version 16.04 there is no problem but with version 18.04 on exactly the same machine there is.

Anyway, thank you very much for your help and work and sorry for opening a ticket for a problem that was more dnsmasq and Multipass than Dokku.
